sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -1066,12 +1075,144 @@ async Task performProcessing()
         ///   load balancing between associated processors.
         /// </summary>
         ///
-        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal the request to cancel the operation.</param>
         ///
         private async Task RunProcessingAsync(CancellationToken cancellationToken)
         {
-            CreateConnection();
-            await Task.Delay(Timeout.Infinite, cancellationToken).ConfigureAwait(false);
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+            // Perform the tasks needed for the main processor loop, encompassing load balancing and management
+            // of processing for claimed partitions.
+            //
+            // Ensure that any errors observed, whether fatal or non-fatal, are surfaced to the exception handler in
+            // a fire-and-forget manner.  The processor does not assume responsibility for observing or surfacing exceptions
+            // that may occur in the handler, so the associated task is intentionally unobserved.
+
+            try
+            {
+                var connection = CreateConnection();
+                await using var connectionAwaiter = connection.ConfigureAwait(false);
+
+                var retryPolicy = Options.RetryOptions.ToRetryPolicy();
+                var cycleDuration = new Stopwatch();
+                var partitionIds = default(string[]);
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    cycleDuration.Restart();
+
+                    try
+                    {
+                        partitionIds = await connection.GetPartitionIdsAsync(retryPolicy, cancellationToken).ConfigureAwait(false);
```

Comment:
This is intentional; the service team is working on a change for "some time in the near future" that will allow dynamic increases to partitions.  While your suggestion would work just fine today, this gives us a bit of future proofing.

Since we're not in a hot path with respect to impacting throughput for processing partitions which are already owned and there's an inherent latency with the load balancer cycles, I think the trade-off is worth it, especially since the query is made over an existing connection/link to the service and doesn't have the overhead of the HTTP handshake to consider.

Reply:
> This is intentional; the service team is looking to allow dynamically increasing the partitions at some point in the not-too-distant future.

Gotcha.  Thanks!